lib/logstash/inputs/rabbitmq/march_hare.rb

```diff
@@ -112,6 +112,12 @@ def consume
       @consumer = @q.build_consumer(:block => true) do |metadata, data|
         @codec.decode(data) do |event|
           decorate(event)
+          event['@metadata']={"routingKey"     => metadata.envelope.routingKey,
+                              "deliveryTag"    => metadata.envelope.deliveryTag,
+                              "redeliver"      => metadata.envelope.redeliver,
+                              "exchange"       => metadata.envelope.exchange,
+                              "amqpTimestamp"  => Time.at(metadata.properties.timestamp.getTime/1000),
+                              "consumerTag"    => metadata.consumer_tag}
```

Comment:
Why not include all standard properties while you're at it? You've got user-id, content-type, content-encoding, app-id, and a few others. I'd also consider putting all values in a subfield of `@metadata` to reduce the risk of field name clashes (that way you don't need the "amqp" prefix to the timestamp).

(I don't know how picky the Elastic folks are, but normally the assignment operator has one space on each side.)